lib/controller/settings_screen_controller.dart

```diff
@@ -1,26 +1,65 @@
+import 'package:flutter/material.dart';
 import 'package:get/get.dart';
+import 'package:intl/intl.dart';
 import 'package:introspection_note_mvp/data/repositories/settings_repository.dart';
+import 'package:introspection_note_mvp/data/sharedpref/SharedPreferenceHelper.dart';
+import 'package:introspection_note_mvp/util/notification_util.dart';
+import 'package:permission_handler/permission_handler.dart';
 
-class SettingsScreenController extends GetxController {
+class SettingsScreenController extends GetxController
+    with WidgetsBindingObserver {
   final SettingsRepository repository;
-  SettingsScreenController({required this.repository});
+  final NotificationUtil notificationUtil;
+  SettingsScreenController({
+    required this.repository,
+    required this.notificationUtil,
+  });
 
   final _isLoading = false.obs;
+  final _grantedNotificationPermission = false.obs;
+  final _grantedExactAlarmPermission = false.obs;
   final _enabledRemindNotification = false.obs;
+  final _remindTime = "".obs;
   bool get isLoading => _isLoading.value;
+  bool get grantedNotificationPermission =>
+      _grantedNotificationPermission.value;
   bool get enabledRemindNotification => _enabledRemindNotification.value;
+  bool get grantedExactAlarmPermission => _grantedExactAlarmPermission.value;
+  String get remindTime => _remindTime.value;
 
   @override
   void onInit() {
     super.onInit();
     _initializeData();
+    WidgetsBinding.instance.addObserver(this);
+  }
+
+  @override
+  void onClose() {
+    WidgetsBinding.instance.removeObserver(this);
+    super.onClose();
   }
 
   void _initializeData() async {
     try {
       _isLoading.value = true;
       _enabledRemindNotification.value =
           await repository.getEnableRemindNotification();
+      _grantedNotificationPermission.value =
+          await Permission.notification.status == PermissionStatus.granted;
+      _grantedExactAlarmPermission.value =
+          await Permission.scheduleExactAlarm.status ==
+          PermissionStatus.granted;
+      SharedpreferenceHelper.instance
+          .getString(SharedpreferenceHelper.SETTING_PUSH_NOTIFICATION_TIME)
+          .then((value) {
+            if (value == "") {
+              value = "20:00";
+              _remindTime.value = value;
+            } else {
+              _remindTime.value = value;
+            }
+          });
```

Comment:
ここの`then`って特にこれで書く理由がないなら`await`で書いた方がシンプルで良さそう

```suggestion
      final value = await SharedpreferenceHelper.instance.getString(
        SharedpreferenceHelper.SETTING_PUSH_NOTIFICATION_TIME,
      );
      if (value == "") {
        _remindTime.value = "20:00";
      } else {
        _remindTime.value = value;
      }
```